### ISSUE TITLE
hassio_supervisor armhf 118 fails to start

### ISSUE THREAD

**Issue by lsilva171** [home-assistant/hassio-build#101]
Yesterday, I successfully deployed Hassio on my Odroid XU-4 running Ubuntu LTS 16.04 but after reinstalling it today the hassio_supervisor contained is now failing to start.

I run the following command to deploy hassio on my Odroid's Docker CE:

curl -sL https://raw.githubusercontent.com/home-assistant/hassio-build/master/install/hassio_install | bash -s -- -m raspberrypi3

The hassio_supervisor container fails with the following error:

Traceback (most recent call last):,
OverflowError: Python int too large to convert to C long,
    "__main__", mod_spec),
  File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main,
  File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code,
    exec(code, run_globals),
  File "/usr/local/lib/python3.7/site-packages/hassio/__main__.py", line 24, in <module>,
    attempt_use_uvloop(),
  File "/usr/local/lib/python3.7/site-packages/hassio/__main__.py", line 15, in attempt_use_uvloop,
    import uvloop,
    from .loop import Loop as __BaseLoop  # NOQA,
  File "/usr/local/lib/python3.7/site-packages/uvloop/__init__.py", line 7, in <module>,
  File "uvloop/includes/stdlib.pxi", line 128, in init uvloop.loop

All the necessary packages and requirements seem to be in place.

Any tips about how to troubleshoot this error?

**Comment by endless1651** [comment]
remove -- -m raspberrypi3 ,and try again

**Comment by lsilva171** [comment]
Thanks for helping me with this issue.

Removing -m will be a problem since hassio_install script has the following statement:

```
case $ARCH in
    "i386" | "i686")
        MACHINE=${MACHINE:=qemux86}
        HOMEASSISTANT_DOCKER="$DOCKER_REPO/$MACHINE-homeassistant"
        HASSIO_DOCKER="$DOCKER_REPO/i386-hassio-supervisor"
    ;;
    "x86_64")
        MACHINE=${MACHINE:=qemux86-64}
        HOMEASSISTANT_DOCKER="$DOCKER_REPO/$MACHINE-homeassistant"
        HASSIO_DOCKER="$DOCKER_REPO/amd64-hassio-supervisor"
    ;;
    "arm" | "armv7l" | "armv6l")
        if [ -z $MACHINE ]; then
            echo "[ERROR] Please set machine for $ARCH"
            exit 1
        fi
        HOMEASSISTANT_DOCKER="$DOCKER_REPO/$MACHINE-homeassistant"
        HASSIO_DOCKER="$DOCKER_REPO/armhf-hassio-supervisor"
    ;;
    "aarch64")
        if [ -z $MACHINE ]; then
            echo "[ERROR] Please set machine for $ARCH"
            exit 1
        fi
        HOMEASSISTANT_DOCKER="$DOCKER_REPO/$MACHINE-homeassistant"
        HASSIO_DOCKER="$DOCKER_REPO/aarch64-hassio-supervisor"
    ;;
    *)
        echo "[Error] $ARCH unknown!"
        exit 1
    ;;
esac
```

So in my case, being armv7l, I need to enter one the available machine types (either armhf or raspberry3).

It shouldn't impact getting the hassio-superviser container (where this issue seems to be) as there's only one but it instruct hassio_install to download the right homeassistant container.

So, under the light of the above statements, will the need to use -m clause impact your fix at all?

**Comment by pvizeli** [comment]
That was a uvloop bug that is fixed with 0.11.0 and that is installed with >= 119
